ray-curator/pyproject.toml

```diff
@@ -15,21 +15,25 @@ requires-python = ">=3.10,<3.13"
 dependencies = [
     "ray[default]",
     "fsspec",
-    "cudf-cu12==25.6.*",
-    "cuml-cu12==25.6.*",
-    "pylibraft-cu12==25.6.*",
-    "raft-dask-cu12==25.6.*",
-    "cugraph-cu12==25.6.*",
-    "nx-cugraph-cu12==25.6.*",
     "torch",
     "loguru",
-    "crossfit @ git+https://github.com/rapidsai/crossfit.git@main",
+    "crossfit>=0.0.9",
     "cosmos-xenna @ git+https://github.com/nvidia-cosmos/cosmos-xenna.git@main",
     "transformers>=4.48.0",
     "pyarrow",
+    "absl-py>=2.0.0,<3.0.0",
```

Comment:
Needed to add this because of some weird pip dependency resolution issues for `tensorflow-metadata` when installing ray and CrossFit in a python 3.10 environment. 
Link to test run: https://github.com/NVIDIA-NeMo/Curator/actions/runs/15720249415/job/44299534230

Longer term we should explore removing a lot of additional CrossFit dependencies that are not needed by default (tf, tf-metadata etc).

Reply:
Yes! Also interestingly https://github.com/huggingface/transformers/pull/38758